filter-parser/src/lib.rs

```diff
@@ -70,7 +71,7 @@ const MAX_FILTER_DEPTH: usize = 200;
 #[derive(Debug, Clone, Eq)]
 pub struct Token<'a> {
     /// The token in the original input, it should be used when possible.
-    span: Span<'a>,
+    pub span: Span<'a>,
```

Comment:
It feels a bit "strange" to me to expose this field publicly, as it feels to me like this struct has an invariant of "uses the original span", unless its value was afterwards modified.

It doesn't feel right for me to be able to mutate the span if e.g. we already have a `value` to `Some`.

Should we consider an `original_span` getter instead?

Reply:
OK I saw your :+1: tamo so I implemented this change in a new commit.

Reply:
Noice